chore(docs): re-structuring how-to-guides

signorecello opened the pull request:
- Moving the token contract tutorial to its place (removing the old one which was already a draft) and renaming the file correctly
- Making all the how-to-guides generic. Snippets are not using #include_code anymore but are generic, i.e. use another_contract::another_function to explain how to call another function from another contract. This helps the distinction between how-tos and tutorials, plus makes them more maintainable assuming no big changes to the syntax.
- Adding some structure to the how to guides so it's easier to just add new ones :)
- Fixing broken links
- Making the guides work with the new Wallet interface

critesjosh commented on docs/docs/developers/getting_started/getting_started.md:
Was the token contract tutorial moved somewhere, or removed? It looks like this is being changed back to "getting started on sandbox" tutorial and I'm not sure where the existing content is. Should it be added to the contract tutorial section?

critesjosh replied:
chatted on slack, update incoming

critesjosh commented:
> makes them more maintainable assuming no big changes to the syntax

big assumption. let's keep an eye on this

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_use_authwit.md:
````diff
@@ -0,0 +1,149 @@
+Check if the current call is authenticated using `assert_current_call_valid_authwit`:
+
+```rust
+// docs:start:authorize_once
````
remove docs comments from code snippets

critesjosh replied:
also the code snippet doesnt include `assert_current_call_valid_authwit`

signorecello replied:
Oh thanks, it does internally but I was supposed to document the macro here. Will fix

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_use_authwit.md:
```diff
@@ -0,0 +1,149 @@
+
+### Validate authentication in a public function
+
+Use `assert_current_call_valid_authwit_public` for public function authentication:
```
the following code snippet does not use this call either

critesjosh replied:
is this checked by the `authorize_once` macro? if so, this should be made more clear.

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_use_authwit.md:
````diff
@@ -0,0 +1,149 @@
+}
+```
+
+### Create the authentication witness in TypeScript
````
At the beginning you mention to check the JS page for JS interaction, but then it's also included here. maybe just keep the contract interaction patterns here

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
```diff
@@ -1,83 +1,118 @@
+### Access storage in functions
 
-You can have multiple `map`s in your contract that each have a different underlying note type, due to note type IDs. These are identifiers for each note type that are unique within a contract.
+Use the `storage` keyword to access your storage variables in contract functions. The `Context` parameter provides execution mode information.
```
>  The `Context` parameter provides execution mode information.

Maybe move this part above the `### Access storage in functions`, since its relevant for the Storage struct definition but not so much for "accessing storage in functions"

signorecello replied:
Yep definitely

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
```diff
@@ -1,83 +1,118 @@
-#include_code private_map /noir-projects/noir-contracts/contracts/app/nft_contract/src/main.nr rust
+- Keys: Always `Field` or serializable types
+- Values: Any type, including other maps
+- Storage: Specified by `storage_slot` and type constructor
```
developers don't need to specify the storage slot though right? maybe best to remove

signorecello replied:
Yeah this little explanation kinda implies they have. Best to remove it and leave explanations to their section (once I start working on it)

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
```diff
@@ -1,83 +1,118 @@
+- Keys: Always `Field` or serializable types
+- Values: Any type, including other maps
+- Storage: Specified by `storage_slot` and type constructor
+- Multiple maps: Supported with unique note type IDs
```
have we talked about note type IDs anywhere else?

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
```diff
@@ -1,83 +1,118 @@
+### Access map values
 
-Above, we are specifying that we want to get the storage in the Map `at` the `msg_sender()`, read the value stored and check that `msg_sender()` is indeed a minter. Doing a similar operation in Solidity code would look like:
+Use the `::at` method to access values by key:
```
```suggestion
Use the `.at()` method to access values by key:
```

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
```diff
@@ -157,7 +201,17 @@ For example, if the storage slot depends on the an address then it is possible t
 
 Set the value of an PrivateImmutable by calling the `initialize` method:
 
-#include_code initialize-private-mutable /noir-projects/noir-contracts/contracts/docs/docs_example_contract/src/main.nr rust
```
we should clean up the `include_code` comments when these are removed, otherwise the codebase is going to be littered with these comments that aren't doing anything

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
````diff
@@ -191,17 +250,21 @@ Functionally similar to `get_note`, but executed unconstrained and can be used b
 
-#include_code insert /noir-projects/aztec-nr/easy-private-state/src/easy_private_uint.nr rust
+```rust
+self.set.insert(addend_note).emit(encode_and_encrypt_note(self.context, owner));
````
i dont think we use this `self.set` syntax anywhere except in aztec.nr, so probably not the best example. i think this is more obvious and reflective of how it should
```suggestion
storage.set.at(aztec_address).insert(new_note).emit(encode_and_encrypt_note(&mut context, aztec_address));
```

signorecello replied:
Claude hallucinations indeed...

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
````diff
@@ -225,7 +288,10 @@ Because of this limit, we should always consider using the second argument `Note
 
-#include_code pop_notes /noir-projects/aztec-nr/easy-private-state/src/easy_private_uint.nr rust
+```rust
+let options = NoteGetterOptions::with_filter(filter_notes_min_sum, subtrahend as Field);
````
it's not clear what should be passed as params here. the with_filter function takes a function and filter parameters. the `filter_notes_min_sum` is imported, it may be worth noting that the value_note lib comes with filter functions.

signorecello replied:
Yes just took this from the old reference without defining it at least. Will fix

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
I can't comment on the line specifically, but `insert_from_public` doesnt exist in PrivateSet anymore and should be removed

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
```diff
@@ -241,101 +307,135 @@ Note that if you obtained the note you are about to remove via `get_notes` it's
-
-For a version of `PublicMutable` that can also be read in private, head to [`DelayedPublicMutable`](#delayed-public-mutable).
+- Generic over any serializable type `T`
+- Uses `storage_slot` to determine storage location
```
storage slot is automatically derive using the storage macro so i think we can remove this point

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_storage.md:
Mike recently added a lot of great comments to the aztec.nr source code that may be helpful reference for updates to this section. Just flagging in case you hadn't checked before editing this page. https://github.com/AztecProtocol/aztec-packages/tree/next/noir-projects/aztec-nr/aztec/src/state_vars

signorecello replied:
Thanks. Would be too much at this point but very useful once we flesh out the reference with eng

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_emit_event.md:
```diff
@@ -2,56 +2,81 @@
+### Choose encryption method
 
-You can find the implementation of event logging [here](https://github.com/AztecProtocol/aztec-packages/blob/#include_aztec_version/noir-projects/aztec-nr/aztec/src/messages/logs/event.nr)
+- `encode_and_encrypt_event`: Constrained encryption, guarantees correct recipient
```
is this right? searching for this method in the repo returns 0 hits

signorecello replied:
Hmm seems like it has been deprecated indeed, it existed on the previous guide so didn't really check it

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_implement_custom_notes.md:
```diff
@@ -0,0 +1,493 @@
+
+// We derive the Serialize trait because this struct is returned from a contract function
+// We use Eq for comparisons, Packable for packing/unpacking the note
+#[derive(Eq, Serialize, Deserialize, Packable)]
```
if we use this as an example i think we should explain what all of these things mean. Maybe there is a simpler example we could use?

signorecello replied:
Yeah this is definitely either not generic enough, or should have some context

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_implement_custom_notes.md:
````diff
@@ -0,0 +1,493 @@
+For efficient lookups by specific fields:
+
+```rust
+#[note]
````
just curious if you sanity checked these examples by putting them into contracts and tested them to ensure they work as expected, or if they were pulled from code references?

signorecello replied:
Claude got them from the code but I'll test them

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,286 @@
+await contract.methods.my_function(param1)
+    .send({ 
+        from: senderAddress,
+        gasSettings: {
```
I dont see `gasSettings` as an option on the SendMethodOptions. we should probably add info about the other options as well https://github.com/AztecProtocol/aztec-packages/blob/eeb8a486f58c2ca995b929199c88fdfbf4f73e00/yarn-project/aztec.js/src/contract/interaction_options.ts#L21

signorecello replied:
It exists but under `fee`, there's a missing object here. Will fix and add others

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,286 @@
+
+## Handle public transactions
+
+### Execute public function
```
Is it worth specifying `contract.methods.private_function`, `contract.methods.private_to_public_function` and `contract.methods.public_function`? they are all the same syntax

signorecello replied:
Yeah I'll just merge these

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,286 @@
+import { BatchCall } from '@aztec/aztec.js';
+
+const batch = new BatchCall(wallet, [
+    contract.methods.function1(param1).request(),
```
looking through the source code, i dont see the need for `.request()`.

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,286 @@
+const tx = contract.methods.my_function(param1).send({ from: senderAddress });
+
+// Monitor status
+tx.onMined((receipt) => {
```
also not sure that this syntax is correct

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
````diff
@@ -0,0 +1,286 @@
+
+```typescript
+// Get current nonce
+const nonce = await pxe.getAccountNonce(senderAddress);
````
i dont see this method in source

critesjosh reviewed (commented):
It looks like Claude was hallucinating with some if the generated docs. Did you review everything it wrote? 

I think it'd also be good to get @sklppy88 to review some of these guides as well since he is familiar with much of the included content.

I didn't review everything yet, but wanted to submit what I have reviewed since it will need some additional attention. I will keep going through this today

signorecello commented:
I confess that I wasn't as thorough as you were in checking for hallucinations. Turns out they're a lot and I should stop being lazy.

No luck in having Claude check for his own hallucinations either so yes, we need human work like in the days of old

critesjosh commented on docs/docs/developers/guides/smart_contracts/structure_project.md:
Why delete this file?

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,93 @@
+yarn add @aztec/aztec.js@#include_version_without_prefix
+```
+
+## Create a PXE client
````
Is this the recommended way to do this? I think the [way we do it in aztec-starter](https://github.com/AztecProtocol/aztec-starter/blob/main/src/utils/setup_pxe.ts) is the way we should be recommending since it separates the notions of the network and the PXE properly. I know this might change soon, but we should publish the correct way

critesjosh replied:
I guess the downside of doing it the recommended way is that it is much more complicated to get started using accounts.

signorecello replied:
I didn't actually change from the original guide [here](https://github.com/AztecProtocol/aztec-packages/blob/5e3219ae64ae2be600d433402de436f714ee0026/docs/docs/developers/guides/js_apps/connect_to_sandbox.md)

Let's discuss async, I'd rather keep the guide shorter and explain less about this separation of concerns

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,93 @@
+
+```typescript
+const nodeInfo = await pxe.getNodeInfo();
+console.log('Connected to sandbox version:', nodeInfo.nodeVersion);
````
```suggestion
logger.info('Connected to sandbox version:', nodeInfo.nodeVersion);
```

signorecello replied:
Well I want this log to show on the user's console, not on the sandbox logs. Actually I'll just remove the needless logger above

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,93 @@
+```typescript
+const nodeInfo = await pxe.getNodeInfo();
+console.log('Connected to sandbox version:', nodeInfo.nodeVersion);
+console.log('Chain ID:', nodeInfo.l1ChainId);
````
```suggestion
logger.info('Chain ID:', nodeInfo.l1ChainId);
```

signorecello replied:
Same as above, I'd rather KISS and use console.log all the way

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
```diff
@@ -0,0 +1,93 @@
+const accounts = await getDeployedTestAccountsWallets(pxe);
+const [aliceWallet, bobWallet] = accounts;
+
+console.log(`Alice's address: ${aliceWallet.getAddress()}`);
```
use `logger.info`

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_create_account.md:
````diff
@@ -0,0 +1,109 @@
+```typescript
+import { getDeployedTestAccountsWallets } from '@aztec/accounts/testing';
+
+const fundedWallet = (await getDeployedTestAccountsWallets(pxe))[0];
````
I think we should show how to do this on testnet as well. aztec-starter does this on the testnet branch https://github.com/AztecProtocol/aztec-starter/blob/testnet/src/utils/deploy_account.ts

signorecello replied:
Yes I'll add instructions for the testnet too

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,285 @@
+        .then(req => req.calls[0].args[0])
+);
+
+console.log(`Contract will deploy at: ${address}`);
```
use logger.info?

signorecello replied:
Logger doesn't exist on this how-to, I'd rather KISS here

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,285 @@
+// First register the contract class if not already registered
+await pxe.registerContractClass(artifact);
+
+// To register an existing contract instance, you need to know
```
Above, only the contract class is registered.

The comment makes me think a step is missing here. Don't we need to register the contract in the pxe with something like this?

```ts
await pxe.registerContract({ instance: sponsoredFPC, artifact: SponsoredFPCContract.artifact });
```

signorecello replied:
Yes definitely, didn't add this step

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,285 @@
+import { loadContractArtifact, ContractInstanceWithAddress } from '@aztec/aztec.js';
+
+// Load the contract artifact
+const artifact = loadContractArtifact(MyContractArtifact);
```
Where does the artifact come from? I think it'd be worth mentioning that it's in the `target/`

signorecello replied:
It's already a few sections imported above, I'm assuming the how-to is small enough that people will read all of it in sequence

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
```diff
@@ -1,284 +1,178 @@
-| Fee Juice (from another account) | New account deployment | Another funded account |
 | Fee Juice (bridged) | Initial funding | L1 assets |
-| Custom FPC | Pay with other tokens | FPC setup |
+| Sponsored FPC | Testing/sandbox bootstrap | Sandbox environment |
```
Sponsored FPC is set up on the testnet is well, so one has to mess with fee juice on testnet

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
```diff
@@ -1,284 +1,178 @@
+:::
 
-## Payment Methods Overview
+## Payment methods overview
```
The table isn't very clear. Since there is more detail in the sections below, it may be worth removing it

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
````diff
@@ -1,284 +1,178 @@
-#include_code claim_and_deploy yarn-project/bot/src/factory.ts javascript
+```typescript
+import { PrivateFeePaymentMethod } from '@aztec/aztec.js';
+import { FPCContract } from "@aztec/noir-contracts.js/FPC";
````
```suggestion
```
this import isnt used in the snippet

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
````diff
@@ -1,284 +1,178 @@
 ```
 
-#### Public Payment
+### Sponsored payments (sandbox only)
````
```suggestion
### Sponsored payments
```
this works on testnet as well

critesjosh replied:
you can see how to set it up for testnet here: https://github.com/AztecProtocol/aztec-starter/blob/testnet/scripts/deploy_contract.ts

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,204 @@
+    contract.methods.updateState()
+]);
+
+const receipt = await batch.send().wait();
```
```suggestion
const receipt = await batch.send({ from: wallet.getAddress() }).wait();
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
````diff
@@ -0,0 +1,204 @@
+### Check transaction effects
+
+```typescript
+const receipt = await sentTx.wait();
````
```suggestion
const effect = await pxe.getTxEffect(tx.txHash);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,204 @@
+const receipt = await sentTx.wait();
+
+// Access public data writes
+receipt.publicDataWrites.forEach(write => {
```
```suggestion
effect.data.publicDataWrites.forEach(write => {
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,204 @@
+});
+
+// Check note hashes (private note commitments)
+receipt.noteHashes.forEach(noteHash => {
```
update this too

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,204 @@
+});
+
+// Check nullifiers (consumed notes)
+receipt.nullifiers.forEach(nullifier => {
```
update this too

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
```diff
@@ -0,0 +1,204 @@
+);
+
+// Compute the message hash for the authorization
+const messageHash = await computeAuthWitMessageHash(aliceWallet, {
```
```suggestion
const messageHash = await computeAuthWitMessageHash({
    caller: wallet.getAddress(),
    action
}, {
    chainId: wallet.getChainId(),
    version: wallet.getVersion(),
});
```
I put this in my IDE and this doesn't give linter errors

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,240 @@
+```typescript
+// Utility functions return properly typed values
+const typedResult = await contract.methods.compute_value(input1, input2)
+    .simulate();
````
```suggestion
    .simulate({ from: wallet.getAddress() });
```
It seems that utility functions require a `from` as well

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
```diff
@@ -0,0 +1,240 @@
+const aggregatedData = await contract.methods.get_aggregated_stats(
+    startBlock,
+    endBlock
+).simulate();
```
Add `from` here too

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
```diff
@@ -0,0 +1,240 @@
+const historicalValue = await contract.methods.get_value()
+    .simulate({
+        from: callerAddress,
+        blockNumber: 12345
```
```suggestion
```

I don't see this option

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
```diff
@@ -0,0 +1,240 @@
+const result = await contract.methods.complex_function(param1)
+    .simulate({
+        from: callerAddress,
+        debug: true
```
I dont see this as an option either

critesjosh commented on docs/docs/developers/guides/smart_contracts/advanced/how_to_prove_history.md:
````diff
@@ -0,0 +1,87 @@
+## Retrieve notes for proofs
+
+```rust
+// Get notes from storage with default options
````
maybe add this line

```
use aztec::note::note_getter_options::NoteGetterOptions,
```

critesjosh reviewed (changes requested):
Chunking my review since it's taking a while :)

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,240 @@
+### Step 2: Handle return values
+
+```typescript
+// Public functions return Field array of size 4
````
this makes it sound like all public functions return 4 Fields

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
````diff
@@ -28,40 +28,70 @@ You can use `aztec.js` to write assertions about transaction statuses, about cha
-#include_code create_pxe_client /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+```typescript
+const PXE_URL = process.env.PXE_URL || 'http://localhost:8080';
+const pxe = createPXEClient(PXE_URL);
````
it would be good to show how to do this on testnet as well. Since this shows up in a few places, it might be worth making its own page.

signorecello replied:
I think it's good to have the guides be generic on both situations, since it tells the reader "how to do something". We don't necessarily want users to have to read another page with tons of irrelevant info just to know how to do one specific thing on the testnet

I'll add a little paragraph on how to do this stuff on the testnet

signorecello replied:
Actually as I look at it, seems like I'm duplicating a ton of info from the getting started guide. This guide should be more straight to the point. A new reader would have to jump through multiple pages if they don't have a PXE yet but that's not a concern of the guides (that's the purpose of the learning journey). Also LLMs would fill up their context with unrelated stuff

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_create_account.md:
How do I create a wallet, not in the sandbox? What do i need to import from aztec.js? I am trying to do this in aztec-starter right now and it's not clear

signorecello replied:
Hmmm I see what you mean. You need to use the Sponsored FPC (or bridge fee juice but we're not documenting that... yet).

I'll make this guide clearer on that part

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
```diff
@@ -28,40 +28,70 @@ You can use `aztec.js` to write assertions about transaction statuses, about cha
+
+// Access transaction effects for debugging
+const txEffects = await pxe.getTxEffect(tx.txHash);
+console.log('New note hashes:', txEffects.noteHashes);
```
```suggestion
console.log('New note hashes:', txEffects.data.noteHashes);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
```diff
@@ -28,40 +28,70 @@ You can use `aztec.js` to write assertions about transaction statuses, about cha
+// Access transaction effects for debugging
+const txEffects = await pxe.getTxEffect(tx.txHash);
+console.log('New note hashes:', txEffects.noteHashes);
+console.log('New nullifiers:', txEffects.nullifiers);
```
```suggestion
console.log('New nullifiers:', txEffects.data.nullifiers);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
```diff
@@ -28,40 +28,70 @@ You can use `aztec.js` to write assertions about transaction statuses, about cha
+const txEffects = await pxe.getTxEffect(tx.txHash);
+console.log('New note hashes:', txEffects.noteHashes);
+console.log('New nullifiers:', txEffects.nullifiers);
+console.log('Public data writes:', txEffects.publicDataWrites);
```
```suggestion
console.log('Public data writes:', txEffects.data.publicDataWrites);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
```diff
@@ -71,29 +101,66 @@ You can also log directly from Aztec contracts. Read [this guide](../local_env/h
+expect(receipt.status).toEqual(TxStatus.APP_LOGIC_REVERTED);
+
+// Verify state wasn't modified
+const ownerPublicBalanceSlot = await cheats.aztec.computeSlotInMap(
```
I dont think we described how to set up `cheats` yet. Oh I see it is below, it may be worth bringing that section up higher in this doc given cheat codes are referenced

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
````diff
@@ -127,7 +235,22 @@ You can check the logs of events emitted by contracts. Contracts in Aztec can em
+expect(logs[0].log.getEmittedFields()).toEqual([value]);
+```
 
 ## Cheats
````
Maybe move this section up

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,97 @@
+Although the sandbox comes with its own PXE, it's useful to create one specifically for your use-case. You will need to bring your own PXE to the testnet eventually. Let's create a PXE store and configure it:
+
+```typescript
+const config = getPXEServiceConfig()
````
should we indicate the import for `getPXEServiceConfig`?

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
```diff
@@ -0,0 +1,97 @@
+const fullConfig = { ...config, l1Contracts }
+fullConfig.proverEnabled = false; // you'll want to set this to "true" once you're ready to connect to the testnet
+
+const store = await createStore('pxe', {
```
also where to import this from?

signorecello replied:
It's very messy because now we have three imports from three different places... Yikes. Anyway, Fixed

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
```diff
@@ -0,0 +1,97 @@
+    dataDirectory: 'store',
+    dataStoreMapSizeKB: 1e6,
+});
+const pxe = await createPXEService(node, fullConfig, {store});
```
and this one

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
```diff
@@ -28,40 +28,70 @@ You can use `aztec.js` to write assertions about transaction statuses, about cha
 
 Import `aztecjs`. This is an example of some functions and types you might need in your test:
```
```suggestion
Import `aztec.js`. This is an example of some functions and types you might need in your test:
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,97 @@
+```typescript
+import { getInitialTestAccountsData } from '@aztec/accounts/testing';
+
+const [aliceAccount, bobAccount] = await getInitialTestAccountsData();
````
called aliceAccount and bobAccount here, but alice and bob below

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,97 @@
+await wallet.createSchnorrAccount(bob.secret, bob.salt);
+```
+
+### Step 2: Check account balances
````
```suggestion
### Check account balances
```

There is no "step 1" above

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_connect_to_sandbox.md:
````diff
@@ -0,0 +1,97 @@
+```typescript
+import { getFeeJuiceBalance } from '@aztec/aztec.js';
+
+const aliceBalance = await getFeeJuiceBalance(aliceWallet.address, pxe);
````
```suggestion
const aliceBalance = await getFeeJuiceBalance(aliceAccount.address, pxe);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_create_account.md:
````diff
@@ -0,0 +1,85 @@
+```typescript
+// you would have something like:
+const [fundedAccount] = await getInitialTestAccountsData();
+await wallet.createSchnorrAccount(fundedAccount.secret, fundedAccount.salt);
````
```suggestion
// add the funded account to the wallet
await wallet.createSchnorrAccount(fundedAccount.secret, fundedAccount.salt);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_create_account.md:
```diff
@@ -0,0 +1,85 @@
+const [fundedAccount] = await getInitialTestAccountsData();
+await wallet.createSchnorrAccount(fundedAccount.secret, fundedAccount.salt);
+
+const alice = await wallet.createSchnorrAccount(secretKey, salt, signingPrivateKey);
```
```suggestion
// add the new account to the wallet
const alice = await wallet.createSchnorrAccount(secretKey, salt, signingPrivateKey);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_create_account.md:
```diff
@@ -0,0 +1,85 @@
+await wallet.createSchnorrAccount(fundedAccount.secret, fundedAccount.salt);
+
+const alice = await wallet.createSchnorrAccount(secretKey, salt, signingPrivateKey);
+await alice.deploy({ deployAccount: fundedAccount.address }).wait()
```
```suggestion
// deploy the new account from the funded account
await alice.deploy({ deployAccount: fundedAccount.address }).wait()
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_create_account.md:
````diff
@@ -0,0 +1,85 @@
+Or if using the Sponsored FPC:
+
+```typescript
+const alice = await wallet.createSchnorrAccount(secretKey, new Fr(0), signingPrivateKey);
````
```suggestion
const alice = await wallet.createSchnorrAccount(secretKey, salt, signingPrivateKey);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+```typescript
+// Deploy with constructor arguments
+await MyContract.deploy(
+    some_wallet,
````
```suggestion
    deployer_wallet,
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+    constructorArg1,
+    constructorArg2
+)
+    .send({ from: testAccount.address }) // testAccount has fee juice
```
The `from` account is required to be an account registered in the wallet that is passed as the first arg to the `deploy` function, correct? It might be good to explicitly state this.

signorecello replied:
Yeah good call

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+    .deployed()
+```
+
+:::
````
```suggestion
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+
+## Use deployment options
+
+### Deploy with custom salt
```
Add a note that the default deployment salt is random, unless specified.

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+
+const contract = await MyContract.deploy(wallet, arg1, arg2)
+    .send({
+        from: wallet.address,
```
the wallet is not tied to 1 address
```suggestion
        from: testAccount.address,
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+```typescript
+const contract = await MyContract.deploy(wallet, arg1, arg2)
+    .send({
+        from: wallet.address,
````
```suggestion
        from: testAccount.address,
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+```typescript
+const contract = await MyContract.deploy(wallet)
+    .send({
+        from: wallet.address,
````
```suggestion
        from: testAccount.address,
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+
+// Initialize later
+await contract.methods.initialize(arg1, arg2)
+    .send({ from: wallet.address })
```
```suggestion
    .send({ from: testAccount.address })
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+import { Fr } from '@aztec/aztec.js';
+
+const salt = Fr.random();
+const deployer = wallet.address;
```
```suggestion
const deployer = testAccount.address;
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+
+```typescript
+const deployTx = MyContract.deploy(wallet, arg1, arg2)
+    .send({ from: wallet.address });
````
```suggestion
    .send({ from: testAccount.address });
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+// Deploy first contract
+const token = await TokenContract.deploy(
+    wallet,
+    wallet.address,
```
```suggestion
    testAccount.address,
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+    wallet,
+    token.address  // Pass first contract's address
+)
+    .send({ from: wallet.address })
```
```suggestion
    .send({ from: testAccount.address })
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+```typescript
+// Start all deployments simultaneously
+const deployments = [
+    Contract1.deploy(wallet, arg1).send({ from: wallet.address }),
````
```suggestion
    Contract1.deploy(wallet, arg1).send({ from: testAccount.address }),
```
and the following 2 lines

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+```
+
+:::tip
+Parallel deployment is faster but be aware of nonce management if deploying many contracts from the same account.
````
What additional info needs to be considered around nonce management? afaik, wallets dont currently enforce any sort of ordering or anything.

signorecello replied:
I think this is up to each wallet, but it might be worth checking with Grego later on

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+// Verify contract is registered in PXE
+const contracts = await pxe.getContracts();
+const isRegistered = contracts.some(
+    c => c.equals(contract.address)
```
```suggestion
    c => c.equals(myContractInstance.address)
```
make it more clear that this is a user supplied value

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+}
+
+// Get contract metadata
+const metadata = await pxe.getContractMetadata(contract.address);
```
```suggestion
const metadata = await pxe.getContractMetadata(myContractInstance.address);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
```diff
@@ -0,0 +1,289 @@
+try {
+    // Try calling a view function
+    const result = await contract.methods.get_version()
+        .simulate({ from: wallet.address });
```
```suggestion
        .simulate({ from: testAccount.address });
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_deploy_contract.md:
````diff
@@ -0,0 +1,289 @@
+```
+
+:::warning
+You need the exact deployment parameters (salt, initialization hash, etc.) to correctly register an externally deployed contract.
````
I think its important to show how to create a contract instance from the deployment params

```ts
      const contract = await getContractInstanceFromInstantiationParams(contractArtifact, {
        publicKeys: PublicKeys.default(),
        constructorArtifact: initializer,
        constructorArgs: parameters,
        deployer: from,
        salt,
      });
```

signorecello replied:
Good call, I'll add a snippet here

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
````diff
@@ -0,0 +1,115 @@
+Let's say you've connected to a contract, for example:
+
+```typescript
+const contract = await Contract.at(contractAddress, artifact, wallet);
````
```suggestion
import { Contract } from "@aztec/aztec.js";

const contract = await Contract.at(contractAddress, artifact, wallet);
```

I think its important to show that there is a specific `Contract` class, so this isn't confused with `MyContract` examples shown elsewhere in the guides

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_send_transaction.md:
````diff
@@ -0,0 +1,115 @@
+
+```typescript
+const txHash = await sentTx.getTxHash();
+const effect = await wallet.getTxEffect(txHash); // or pxe.getTxReceipt(txHash);
````
i dont see this method on the wallet interface, but it is available on the pxe interface.
```suggestion
const effect = await pxe.getTxEffect(txHash); // or pxe.getTxReceipt(txHash);
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
```diff
@@ -0,0 +1,237 @@
+
+## Connect to a contract
+
+### Step 1: Import the Contract module
```
When should we be recommending people import the `Contract` class rather than importing the autogenerated artifact?

signorecello replied:
Asking hard questions haha. I think `Contract` when you don't have the artifact, and the artifact when you do.

Which one is best really depends on the env. On the browser it makes sense to use the artifact but ymmv

signorecello replied:
I'll add both here and in the send_transactions guide too

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,237 @@
+
+```typescript
+const result = await contract.methods.get_public_value(param1)
+    .simulate({ from: callerAddress });
````
The `callerAddress` must be registered in the `wallet` used here, `const contract = await Contract.at(contractAddress, ContractArtifact, wallet);`, correct?

signorecello replied:
Yes I believe so, I'll make it more explicit

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,237 @@
+```typescript
+// Utility functions return properly typed values
+const typedResult = await contract.methods.compute_value(input1, input2)
+    .simulate({ from: wallet.address });
````
```suggestion
    .simulate({ from: account.address });
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,237 @@
+### Step 1: Call utility function
+
+```typescript
+// Utility functions return properly typed values
````
Do you have an example of these returning typed values? I tested it but am not seeing results typed (although Claude is telling me they should be)

signorecello replied:
They should be but I see `any` so I'll just remove this section from the time being

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
```diff
@@ -0,0 +1,237 @@
+const aggregatedData = await contract.methods.get_aggregated_stats(
+    startBlock,
+    endBlock
+).simulate({ from: wallet.address });
```
```suggestion
).simulate({ from: account.address });
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,237 @@
+### Simulate at specific blocks
+
+```typescript
+// Simulate at a historical block
````
I don't see anything different about this example that shows how its being used to get historical info

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
```diff
@@ -0,0 +1,237 @@
+
+## Debug simulations
+
+### Enable verbose output
```
I dont see anything different about this call that increases verbosity

signorecello replied:
Seems like Claude just made up a bunch of stuff here and I cleaned it up, not noticing I wasn't actually doing anything different from above.

I'll just remove this noise altogether. Less = more

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_simulate_function.md:
````diff
@@ -0,0 +1,237 @@
+// Check console for detailed execution logs
+```
+
+### Catch simulation errors
````
Do you think we need to include this? it's basic JS functionality

critesjosh reviewed (changes requested):
Submitting comments for now and will continue reviewing.

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+
+Using AuthWitnesses is always a two-part process. This guide shows how to generate and use them, but you still need to set up your contract to accept and authenticate them.
+
+Therefore it is very much recommended to read the `aztec-nr` [guide on authwitnesses](../smart_contracts/how_to_use_authwit.md) before this one.
```
```suggestion
Therefore it is recommended to read the `aztec-nr` [guide on authwitnesses](../smart_contracts/how_to_use_authwit.md) before this one.
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+- Contract with authwit validation (see [smart contract authwits](../smart_contracts/how_to_use_authwit.md))
+- Understanding of [authwit concepts](../../../aztec/concepts/advanced/authwit.md)
+
+Let's also assume functions `some_public_function` and `some_private_function` with the macro `#[authorize_once("from", "authwit_nonce")]`, meaning it will check if:
```
Seems weird that this is in the `Prerequisites` section. Move into the `AuthWits` section below

critesjosh replied:
```suggestion
Let's also assume we have a contract with functions `some_public_function` and `some_private_function` with the macro `#[authorize_once("from", "authwit_nonce")]`, meaning it will check if:
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+const nonce = Fr.random()
+
+// bob creates an authwit that authorizes alice to call the function on his behalf
+const action = contract.methods.some_private_function(bob, alice, 10n, nonce)
```
It's confusing that `alice` is an arg to the function, and also the caller for the authwit. These things are not related and do not have to be. I think it might be helpful to make the 2nd arg a different address to make this more clear.

signorecello replied:
They will very commonly be related, but they don't have to be. So yes I'll make it different which should illustrate it

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+
+:::tip
+
+The nonce is necessary to avoid replay attacks. However, the contract is `smart` enough (pun intended) to allow bob to call the function himself by setting the nonce to `0`.
```
```suggestion
The nonce is necessary to avoid replay attacks. However, the contract is smart enough to allow bob to call the function himself by setting the nonce to `0`.
```
Suggesting removing this just because it seems more professional without it, which is currently specified in the docs CLAUDE.md file

signorecello replied:
Bah couldn't resist a pun here haha

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+
+## Create public authwits
+
+Public authwits mean the authorization is public, so it is its own transaction. You create the authwit just as above, but the wallet needs to make this action authorized in the canonical `AuthRegistry` contract:
```
```suggestion
Public authwits mean the authorization is public, so it requires a transaction. You create the authwit just as above, but the wallet needs to authorize it in the canonical `AuthRegistry` contract:
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+
+## Create arbitrary message authwits
+
+### Step 1: Create inner hash
```
What is an example of a use case for this? It'd be helpful for helping with understanding how to use it.

signorecello replied:
That's an amazing question you're making, considering the fact that I also don't have the slightest idea about the answer.

Just kept it from the original guide so I suggest leaving it and coming back later if it's not relevant to keep documenting

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_use_authwit.md:
```diff
@@ -0,0 +1,122 @@
+const revoked = await authorizerWallet.setPublicAuthWit({
+    caller: executorAddress,
+    action: action
+}, false);
```
```suggestion
}, false).send({ from: account.address });
```
need to send it to the network

critesjosh commented on docs/docs/developers/guides/js_apps/index.md:
```diff
@@ -10,21 +11,27 @@ Aztec.js is a library that provides APIs for managing accounts and interacting w
+import {
+  createPXEClient,
+  waitForPXE,
+  deployAndInitializeTokenContract,
```
This method doesn't exist

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
````diff
@@ -28,40 +28,70 @@ You can use `aztec.js` to write assertions about transaction statuses, about cha
-#include_code imports /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+```typescript
+import { getInitialTestAccountsData } from '@aztec/accounts/testing';
+import { type AccountWallet, AztecAddress, Fr, type PXE, TxStatus, createPXEClient, waitForPXE } from '@aztec/aztec.js';
````
AccountWallet doesn't exist

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_test.md:
```diff
@@ -71,29 +101,66 @@ You can also log directly from Aztec contracts. Read [this guide](../local_env/h
 
 #### A transaction is dropped
 
 We can have private transactions that work fine locally, but are dropped by the sequencer when tried to be included due to a double-spend. In this example, we simulate two different transfers that would succeed individually, but not when both are tried to be mined. Here we need to `send()` the transaction and `wait()` for it to be mined.
```
```suggestion
We can have private transactions that work fine locally, but are dropped by the sequencer when tried to be included due to an existing nullifier. In this example, we simulate two different transfers that would succeed individually, but not when both are tried to be mined. Here we need to `send()` the transaction and `wait()` for it to be mined.
```
Make the reference more generic

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
```diff
@@ -1,284 +1,165 @@
-  }
+// Deploy account for free
+await yourAccount.deploy({
+    fee: { paymentMethod }
```
```suggestion
    fee: { sponsoredPaymentMethod }
```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
```diff
@@ -1,284 +1,165 @@
-  FPCContract.artifact,
-  fpcDeployParams
+// Private FPCs enable fee payments without revealing the payer's identity onchain.
+const paymentMethod = new PrivateFeePaymentMethod(
```
I am getting syntax errors with this, it needs the senderAddress as well

```js
    const paymentMethod = new PrivateFeePaymentMethod(
        paymentContractAddress,
        senderAddress,
        wallet,
    );
 ```

critesjosh commented on docs/docs/developers/guides/js_apps/how_to_pay_fees.md:
```diff
@@ -1,284 +1,165 @@
-
-# Public FPC
---payment method=fpc-public,fpc=$FPC_ADDRESS,asset=$ASSET_ADDRESS
+const paymentMethod = new PublicFeePaymentMethod(
```
this also requires 3 args

critesjosh commented on docs/docs/developers/guides/smart_contracts/advanced/how_to_profile_transactions.md:
```diff
@@ -0,0 +1,255 @@
+const result = await contract.methods
+  .my_function(args)
+  .profile({ 
+    profileMode: 'gates',
```
```suggestion
    from: address,
    profileMode: 'gates',
```

critesjosh commented on docs/docs/developers/guides/smart_contracts/advanced/how_to_profile_transactions.md:
````diff
@@ -0,0 +1,255 @@
+### Step 2: Profile deployment
+
+```javascript
+const deploy = await Contract.deploy(args).profile({ profileMode: 'full' });
````
```suggestion
const deploy = await Contract.deploy(args).profile({ from: address, profileMode: 'full' });
```

critesjosh commented on docs/docs/developers/guides/smart_contracts/advanced/how_to_prove_history.md:
```diff
@@ -0,0 +1,88 @@
+let options = NoteGetterOptions::new();
+let notes = storage.notes.at(owner).get_notes(options);
+
+// Access first note as RetrievedNote
```
```suggestion
// Access first note as retrieved_note
```

critesjosh commented on docs/docs/developers/guides/smart_contracts/advanced/how_to_retrieve_filter_notes.md:
```diff
@@ -0,0 +1,204 @@
+let mut options = NoteGetterOptions::new();
+options = options
+    .select(MyNote::properties().value, Comparator.EQ, value)
+    .select(MyNote::properties().owner, Comparator.EQ, owner);
```
I know you do it above, but it might be helpful to indicate again that readers need to call `get_notes(options)` after applying the selection criteria

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_communicate_cross_chain.md:
````diff
@@ -0,0 +1,196 @@
+Use `message_portal` in your `context` to send messages from L2 to L1:
+
+```rust
+#[public]
````
Might be good to mention that this can be done from a public or a private function

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_functions.md:
````diff
@@ -0,0 +1,123 @@
+}
+```
+
+Private functions maintain privacy of user inputs and execution logic.
````
might be worth noting that private functions only have access to private state

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_define_functions.md:
````diff
@@ -0,0 +1,123 @@
+}
+```
+
+Public functions can access public state, similar to EVM contracts.
````
```suggestion
Public functions can access public state, similar to EVM contracts. Public functions do not have direct access to private state.
```

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_emit_event.md:
````diff
@@ -1,57 +1,87 @@
-- `encode_and_encrypt_event` Sends an encrypted message to `recipient` with the content of the event, which they will discover when processing private logs.
-- `encode_and_encrypt_event_unconstrained` is the same as `encode_and_encrypt_event`, except encryption is unconstrained. This means that the sender is free to make the log contents be whatever they wish, so the recipient is trusting the sender of the event. This could also potentially result in scenarios in which the recipient is unable to decrypt and process the payload, **leading to the event  ...
+```rust
+emit_event_in_private(
````
indicate which package this is imported from

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_emit_event.md:
````diff
@@ -1,57 +1,87 @@
 
-## Public Logs
+```rust
+emit_event_in_public(
````
indicate which package to import from

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_test_contracts.md:
```diff
@@ -0,0 +1,350 @@
+For complex cross-chain or integration testing, see the [TypeScript testing guide](../js_apps/how_to_test.md).
+:::
+
+## Write pure Noir tests
```
This feels like an edge case for testing, I think most readers will want contract testing. maybe move this to the bottom of the page

signorecello replied:
I'll actually remove this, since this is just Noir testing

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_test_contracts.md:
```diff
@@ -0,0 +1,350 @@
+let deployer = env.deploy("ContractName");
+
+// If on a different crate:
+let deployer = env.deploy("../../@token_contract", "Token");
```
I believe this only takes 1 arg. The code comments on the fn are pretty good https://github.com/AztecProtocol/aztec-packages/blob/d0f6018a548e6b9cb196fcdc7f01fe8e3782c783/noir-projects/aztec-nr/aztec/src/test/helpers/test_environment.nr#L457

critesjosh commented on docs/docs/developers/guides/smart_contracts/how_to_use_authwit.md:
````diff
@@ -0,0 +1,96 @@
+1. Compute the message hash using `compute_authwit_message_hash_from_call`
+2. Set the authorization using `set_authorized`
+
+```rust
````
it might be helpful to cite where this code is coming from so people can dig deeper into how it works and cases where they might want to implement something similar. iiuc, this comes from the token bridge code

critesjosh commented on docs/docs/developers/tutorials/contract_tutorials/token_contract.md:
````diff
@@ -0,0 +1,629 @@
+
+Since we're here, let's import more specific stuff from this library:
+
+```noir
````
```suggestion
```rust
```
<img width="1837" height="457" alt="image" src="https://github.com/user-attachments/assets/43a71437-8005-4d6c-b0f6-7ffcf3220f24" />

critesjosh replied:
and for the rest of the file

critesjosh commented on docs/docs/developers/tutorials/contract_tutorials/token_contract.md:
```diff
@@ -0,0 +1,629 @@
+
+1. **Public Layer**: Giggle mints tokens publicly - transparent and auditable
+2. **Private Layer**: Employees transfer and spend tokens privately - completely confidential
+3. **Cross-layer Bridge**: Employees can move tokens between public and private domains as needed
```
```suggestion
3. **Cross-layer Transfer**: Employees can move tokens between public and private domains as needed
```
Bridge means something specific in regards to moving across chains so may be best to avoid the word

critesjosh commented on docs/docs/developers/tutorials/contract_tutorials/token_contract.md:
````diff
@@ -0,0 +1,629 @@
+Then we will use the `giggleWallet` to deploy our contract, mint 100 BOB to Alice, then transfer 10 of those to Bob's Clinic publicly... for now. Let's go:
+
+```typescript
+
````
```suggestion
```

critesjosh commented on docs/docs/developers/tutorials/contract_tutorials/token_contract.md:
```diff
@@ -0,0 +1,629 @@
+        .deploy(giggleWallet)
+        .send({ from: giggleWallet.address })
+        .deployed();
+
```
```suggestion
```

critesjosh commented on docs/docs/developers/tutorials/contract_tutorials/token_contract.md:
```diff
@@ -0,0 +1,629 @@
+3. If the check fails, the entire transaction (including the private part) reverts
+4. Only Giggle can successfully mint BOB tokens
+
+## Part 5: Converting Back to Public
```
```suggestion
## Part 4: Converting Back to Public
```

critesjosh reviewed (commented):
All reviewed! Let's not have a PR this large again haha. Just address the comments and I will approve.

Also, since there are a lot of page renames, it would be good to set up redirects in `netlify.toml`.

signorecello committed:
chore(docs): re-structuring how-to-guides

- Moving the token contract tutorial to its place (removing the old one which was already a draft) and renaming the file correctly
- Making all the how-to-guides generic. Snippets are not using #include_code anymore but are generic, i.e. use another_contract::another_function to explain how to call another function from another contract. This helps the distinction between how-tos and tutorials, plus makes them more maintainable assuming no big changes to the syntax.
- Adding some structure to the how to guides so it's easier to just add new ones :)
- Fixing broken links
- Making the guides work with the new Wallet interface

Co-authored-by: Josh Crites <jc@joshcrites.com>
Co-authored-by: José Pedro Sousa <jose@aztecprotocol.com>
Co-authored-by: José Pedro Sousa <outgoing@zpedro.dev>
